UDV-Core/src/Utils/BaseDemo/js/BaseDemo.js

```diff
@@ -0,0 +1,542 @@
+                break;
+            case BaseDemo.AUTHENTICATION_MODULE:
+                this.createAuthenticationFrame(moduleId);
+                break;
+            default:
+                throw `Unknown module type : ${type}`;
+        }
+
+        if (requireAuth) {
+            this.requireAuthModules.push(moduleId);
+            this.updateAuthentication();
+        }
+    }
+
+    /**
+     * Creates a new button in the side menu.
+     * @param moduleId The module id.
+     * @param buttonText The text to display in the button.
+     */
+    createMenuButton(moduleId, buttonText) {
+        let button = document.createElement('label');
+        button.id = this.getModuleButtonId(moduleId);
+        button.innerText = buttonText;
+        this.menuElement.appendChild(button);
+        let icon = document.createElement('img');
+
+        //creating an icon
+        icon.setAttribute('src', `${this.iconFolder}/${moduleId}.svg`)
+        icon.className = 'menuIcon';
+        button.insertBefore(icon, button.firstChild);
+
+        //define button behavior
+        button.onclick = (() => {
+            this.toggleModule(moduleId);
+        }).bind(this);
+        let moduleClass = this.getModuleById(moduleId);
+
+        //dynamically color the button
+        moduleClass.parentElement = this.contentSectionElement;
+        moduleClass.addEventListener(ModuleView.EVENT_ENABLED, () => {
+            button.className = 'choiceMenu choiceMenuSelected';
+
+        });
+        moduleClass.addEventListener(ModuleView.EVENT_DISABLED, () => {
+            button.className = 'choiceMenu';
+        });
+        moduleClass.disable();
+    }
+
+    /**
+     * Creates an authentication frame for the authentication module.
+     * @param authModuleId The id of the authentication module.
+     */
+    createAuthenticationFrame(authModuleId) {
+        let frame = document.createElement('div');
+        frame.id = this.authenticationFrameId;
+        frame.innerHTML = this.authenticationFrameHtml;
+        this.menuElement.insertBefore(frame,
+            document.getElementById('openHamburger').nextSibling);
+        const authView = this.getModuleById(authModuleId);
+        authView.parentElement = this.contentSectionElement;
+        const authService = authView.authenticationService;
+        this.authenticationLoginButtonElement.onclick = () => {
+            if (this.isModuleActive(authModuleId)) {
+                authView.disable();
+            } else {
+                authView.enable();
+            }
+        };
+        this.authenticationLogoutButtonElement.onclick = () => {
+            try {
+                authService.logout();
+            } catch (e) {
+                console.error(e);
+            }
+        };
+        
+        authService.addObserver(this.updateAuthentication.bind(this));
+        this.authService = authService;
+        this.updateAuthentication();
+    }
+
+    updateAuthentication() {
```

Comment:
Please add documentation like the other functions

Reply:
Added doc comment